source/heroes/ControllableHero.hx

```diff
@@ -0,0 +1,38 @@
+package heroes;
+
+import flixel.FlxSprite;
+import flixel.util.FlxColor;
+import flixel.FlxG;
+
+class ControllableHero extends FlxSprite {
+    var DirectionX:Float;
+    var DirectionY:Float;
+
+    public function new(X:Float=0, Y:Float=0)
+    {
+        super(X, Y);
+        makeGraphic(25, 50, 0xFF006118);
+
+    }
+    override public function update(elapsed:Float)
+    {
+        super.update(elapsed);
+        
+        if(FlxG.keys.pressed.UP)
+        {
+            this.y -=5;
+        }
+        if(FlxG.keys.pressed.DOWN)
+        {
+            this.y +=5;
+        }
+        if(FlxG.keys.pressed.LEFT)
+        {
+            this.x -=5;
+        }
+        if(FlxG.keys.pressed.RIGHT)
+        {
+            this.x +=5;
+        }
```

Comment:
This is nice and functional, but it does fall prey to the problem we discussed in class the other day - `update` calls happen every time the screen is drawn - if we run at a very, very fast framerate our character is going to move 5 pixels for every one of those visual frames.

No change needed here (unless you want to try to solve this problem) we'll discuss it more in class how we can work around the issue.